File: 1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
    const indices = {};
    for (let i = 0; i < nums.length; i++) {
        if (indices[target - nums[i]] >= 0) {
            return [indices[target - nums[i]], i];
        }
        indices[nums[i]] = i;
    }
};

// time complexity: 
// O(N) - iterate through list, getting and setting in an object is O(1)

File: 1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {} 
        for idx, num in enumerate(nums):
            remaining = target - num
            if remaining in seen:
                return [seen[remaining], idx]
            seen[num] = idx
        return []

# time complexity:
# O(N) - iterate through list, getting and setting in a dict is avg O(1)

